src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs

```diff
@@ -134,9 +134,10 @@ public async Task<ODataRequest> CreateLinkRequestAsync(string collection, string
         public async Task<ODataRequest> CreateUnlinkRequestAsync(string collection, string linkName, string entryIdent, string linkIdent)
         {
             var associationName = _session.Metadata.GetNavigationPropertyExactName(collection, linkName);
-            await WriteEntryContentAsync(RestVerbs.Delete, collection, entryIdent, null, false).ConfigureAwait(false);
-
             var commandText = FormatLinkPath(entryIdent, associationName, linkIdent);
+            
+            await WriteEntryContentAsync(RestVerbs.Delete, collection, commandText, null, false).ConfigureAwait(false);
```

Comment:
f you look at the signature of the WriteEntryContentAsync method being called, you can see it expects the commandText variable in the place of the third paramater. In the code as it was before I modified it, the commandText variable is being populated after the call to the WriteEntryContentAsync method was being called. It looked like a typo to me, so I moved the line below and corrected the parameter being passed, then stepped it through and it looked ok. Also inspected the calls in Fiddler and they look correct.

I'm hoping someone else has time to construct a test case as I'm a little pressed for time. However, to reproduce the issue you can have a look at the bug #402. Basically you just need do an unlink (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Updating-entries-with-links#update-a-product-and-unlink-a-category) in a Batch UpdateEntryAsync (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Batch-requests) and you will see this problem.

I've used this fix a lot with Microsoft PowerApps/Dynamics 365 Web API and it works like a charm.